functions/src/test/algoliaSync.test.ts

```diff
@@ -0,0 +1,696 @@
+import * as assert from 'assert'
+import * as sinon from 'sinon'
+import * as mocks from './mocks'
+import 'mocha'
+
+import * as algolia from '../algoliaSync'
+
+describe('Function Handlers', () => {
+  afterEach(() => {
+    // @ts-ignore
+    algolia.algoliaSync.restore()
+  })
+
+  it('Should call algolia sync on changes to activity data', async () => {
+    const spy = sinon.spy(algolia, 'algoliaSync')
+    // @ts-ignore
+    algolia.activityDataHandler(mocks.syncTest.equalChange, {})
+    assert(spy.calledOnce)
```

Comment:
I did a ton of looking through sinon mock functionality, but it never seemed to work how I expected it to - because I often did want the rest of the API to execute "normally". Then again, I probably just never read the right resource for it.